File: PythonSolution/sudokuSolver.py
import sys

f = open(sys.argv[1],'r');

grid = []
concrete = []

for row in f:
	s = row.replace('\n','').split(',')
	grid.append(s)

def solve(cellNo):
	x = cellNo % 9
	y = cellNo // 9
	if int(grid[x][y]) != 0:
		if (cellNo == 80 or solve(cellNo + 1) == True):
			return True
		else:
			return False
	for i in range(1,10):
		if isValid(x,y,i) == True:
			grid[x][y] = i
			if (cellNo == 80 or solve(cellNo + 1) == True):
				return True
	grid[x][y] = 0
	return False

def isValid(x,y,num):
	for i in range(0,9):
		if (int(grid[x][i])==num or int(grid[i][y]) ==num):
			return False
	minx = (x // 3)*3
	miny = (y // 3)*3
	for i in range(minx,minx+3):
		for j in range(miny,miny+3):
			if (int(grid[i][j]) == num):
				return False
	return True

solve(0)
for row in grid:
	for item in row:
		sys.stdout.write(str(item) + ' ')
	sys.stdout.write('\n')

File: HTMLSolution/sudokuSolver.js
var solver = angular.module('sudokuSolver',[]);

var makeCell = function(solid, num){return false};

solver.controller('sudokuController', function($scope){
	var grid = [];
	for (var i = 0; i < 9; i++)
	{
		var row = [];
		for (var j = 0; j < 9; j++)
		{
			row.push({val:null});
		}
		grid.push(row);
	}
	$scope.grid = grid;

	$scope.dsbld = false;

	$scope.clear = function(){
		for (var i = 0; i < 9; i++)
			for (var j = 0; j < 9; j++){
				$scope.grid[i][j].val = null;
			}
			$scope.dsbld = false;
	}
	$scope.do = function()
	{
		$scope.solve(0);
		$scope.dsbld = true;
	}
	$scope.solve = function(cellNo){
		var x = cellNo % 9;
		var y = Math.floor(cellNo / 9);
		if ($scope.grid[x][y].val != null)
		{

			if (cellNo == 80 || $scope.solve(cellNo + 1) == true){
				return true;
			}
			else return false;
		}

		for (var i = 1; i <= 9; i++){
			if ($scope.isValid(x,y,i) == true){

				$scope.grid[x][y].val=i;
				if (cellNo == 80 || $scope.solve(cellNo + 1) == true)
					return true;
				}

		}
		$scope.grid[x][y].val = null;
		return false;
	}
	$scope.isValid = function(x,y,num)
	{
		for (var i = 0; i < 9; i++)
			if ($scope.grid[x][i].val == num || $scope.grid[i][y].val == num)
				return false;
		var minX = (Math.floor(x / 3))*3;
		var minY = (Math.floor(y / 3))*3;

		for (var i = minX; i < minX + 3; i++)
			for (var j = minY; j < minY + 3; j++)
				if ($scope.grid[i][j].val == num)
					return false;

		return true;
	}
});
